plugins/modules/aci_l3out_logical_interface_profile.py

```diff
@@ -308,8 +395,67 @@ def main():
             dict(l3extRsNdIfPol=dict(attributes=dict(tnNdIfPolName=nd_policy))),
             dict(l3extRsIngressQosDppPol=dict(attributes=dict(tnQosDppPolName=ingress_dpp_policy))),
             dict(l3extRsEgressQosDppPol=dict(attributes=dict(tnQosDppPolName=egress_dpp_policy))),
+            dict(l3extRsLIfPCustQosPol=dict(attributes=dict(tnQosCustomPolName=qos_custom_policy))),
         ]
-        aci.payload(aci_class="l3extLIfP", class_config=dict(name=interface_profile, descr=description), child_configs=child_configs)
+        for class_name, attribute in extra_child_classes.items():
+            attribute_input = attribute.get("attribute_input")
+            if attribute_input is not None:
+                rs_class = attribute.get("rs_class")
+                if all(value is None for value in attribute_input.values()) and isinstance(aci.existing, list) and len(aci.existing) > 0:
+                    for child in aci.existing[0].get("l3extLIfP", {}).get("children", {}):
+                        if child.get(class_name):
+                            child_configs.append(
+                                {
+                                    class_name: dict(
+                                        attributes=dict(status="deleted"),
+                                    ),
+                                }
+                            )
+                elif all(value is not None for value in attribute_input.values()):
```

Comment:
the problem with the `required=True` method  is with the way to delete the object. to delete it, the user needs to provide an empty dict but with the previous required method, it will return an error message asking to provide all attributes instead of deleting the object. So we either need to find a new way to delete the object or keep that solution. what do you think?

Reply:
Ah ok I see why you chose this way. This is good.